#! /bin/sh /usr/share/dpatch/dpatch-run
## beagle-crawl-safely.dpatch by Ted Percival <ted@midg3t.net>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Exit if beagle-build-index is not available. Also exit on errors
## DP: (set -e) and redo option-adding to work with "set -e".

@DPATCH@
diff -urNad beagle-0.3.0~/tools/beagle-crawl-system.in beagle-0.3.0/tools/beagle-crawl-system.in
--- beagle-0.3.0~/tools/beagle-crawl-system.in	2007-11-02 11:03:39.000000000 +0100
+++ beagle-0.3.0/tools/beagle-crawl-system.in	2007-11-22 23:33:58.000000000 +0100
@@ -1,4 +1,5 @@
 #!/bin/sh
+set -e
 
 #
 # beagle-crawl-system
@@ -26,6 +27,10 @@
 # SOFTWARE.
 #
 
+if [ ! -x @sbindir@/beagle-build-index ] ; then
+    exit 0
+fi
+
 umask 022
 
 DEBUG=0
@@ -46,7 +51,7 @@
 chown $CRAWL_USER $MONO_SHARED_DIR
 
 for crawl_file in @sysconfdir@/beagle/crawl-*; do
-    unset CRAWL_ENABLED CRAWL_DISABLE_FILTERING CRAWL_RECURSIVE CRAWL_CACHE_TEXT CRAWL_ALLOW_PATTERNS CRAWL_DENY_PATTERNS CRAWL_PATHS CRAWL_INDEX_NAME
+    unset CRAWL_ENABLED CRAWL_DISABLE_FILTERING CRAWL_RECURSIVE CRAWL_CACHE_TEXT CRAWL_ALLOW_PATTERNS CRAWL_DENY_PATTERNS CRAWL_PATHS CRAWL_INDEX_NAME OPTIONS
 
     if [ "$DEBUG" = "1" ]; then
 	echo "$0: Processing files in @sysconfdir@/beagle..."
@@ -59,34 +64,40 @@
 
 	. $crawl_file
 
-	if [ -n "$CRAWL_ENABLED" -a "$CRAWL_ENABLED" = "yes" ] ; then
-
-	    OPTIONS="`test -n "$CRAWL_DISABLE_FILTERING" -a "$CRAWL_DISABLE_FILTERING" = "yes" && \
-                     echo --disable-filtering`"
-
-	    OPTIONS="$OPTIONS `test -n "$CRAWL_DISABLE_DIRECTORIES" -a "$CRAWL_DISABLE_DIRECTORIES" = "yes" && \
-                     echo --disable-directories`"
-
-	    OPTIONS="$OPTIONS `test -n "$CRAWL_DISABLE_ON_BATTERY" -a "$CRAWL_DISABLE_ON_BATTERY" = "yes" && \
-                     echo --disable-on-battery`"
-
-	    OPTIONS="$OPTIONS `test -n "$CRAWL_RECURSIVE" -a "$CRAWL_RECURSIVE" = "yes" && \
-                     echo --recursive`"
-
-	    OPTIONS="$OPTIONS `test -n "$CRAWL_CACHE_TEXT" -a "$CRAWL_CACHE_TEXT" = "yes" && \
-                     echo --enable-text-cache`"
-
-	    OPTIONS="$OPTIONS `test -n "$CRAWL_ALLOW_PATTERNS" && \
-                     echo --allow-pattern $CRAWL_ALLOW_PATTERNS`"
+	if [ "$CRAWL_ENABLED" = "yes" ] ; then
 
-	    OPTIONS="$OPTIONS `test -n "$CRAWL_DENY_PATTERNS" && \
-                     echo --deny-pattern $CRAWL_DENY_PATTERNS`"
+   	    if [ "$CRAWL_DISABLE_FILTERING" = "yes" ] ; then
+   	        OPTIONS="$OPTIONS --disable-filtering"
+   	    fi
+    
+   	    if [ "$CRAWL_DISABLE_DIRECTORIES" = "yes" ] ; then
+   	    	OPTIONS="$OPTIONS --disable-directories"
+   	    fi
+  
+ 	    if [ "$CRAWL_DISABLE_ON_BATTERY" = "yes" ] ; then
+ 	    	OPTIONS="$OPTIONS --disable-on-battery"
+ 	    fi
+ 
+   	    if [ "$CRAWL_RECURSIVE" = "yes" ] ; then
+   	        OPTIONS="$OPTIONS --recursive"
+   	    fi
+    
+    	    if [ "$CRAWL_CACHE_TEXT" = "yes" ] ; then
+    		OPTIONS="$OPTIONS --enable-text-cache"
+    	    fi
+     
+    	    if [ -n "$CRAWL_ALLOW_PATTERNS" ] ; then
+    		OPTIONS="$OPTIONS --allow-pattern $CRAWL_ALLOW_PATTERNS"
+    	    fi
+     
+    	    if [ -n "$CRAWL_DENY_PATTERNS" ] ; then
+    		OPTIONS="$OPTIONS --deny-pattern $CRAWL_DENY_PATTERNS"
+    	    fi
+     
+    	    if IONICE=`which ionice 2>/dev/null`; then
+    		IONICE="$IONICE -c 3"
+    	    fi
 	    
-	    IONICE=`which ionice 2>/dev/null`
-            if [ -n "$IONICE" ]; then
-                IONICE="$IONICE -c 3"
-            fi
-
 	    COMMAND="$IONICE su -s /bin/bash $CRAWL_USER -c \"MONO_SHARED_DIR=$MONO_SHARED_DIR @sbindir@/beagle-build-index --target @localstatedir@/cache/beagle/indexes/$CRAWL_INDEX_NAME $OPTIONS $CRAWL_PATHS\""
 	    
 	    if [ "$DEBUG" = "1" ]; then
